django/core/validators.py

```diff
@@ -65,6 +66,69 @@ def __eq__(self, other):
         )
 
 
+@deconstructible
+class DomainNameValidator(RegexValidator):
+    message = _('Enter a valid domain name.')
+    ul = "\u00a1-\uffff"  # Unicode letters range (must not be a raw string).
+    # Max length for domain name labels is 63 characters per RFC 1034 sec. 3.1
+    domain_re = r"(?:\.(?!-)[a-z" + ul + r"0-9-]{1,63}(?<!-))*"
+    # Top-level domain
+    tld_re = (
+        r"\."  # dot
+        r"(?!-)"  # can't start with a dash
+        r"(?:[a-z" + ul + "-]{2,63}"  # domain label
+        r"|xn--[a-z0-9]{1,59})"  # or punycode label
+        r"(?<!-)"  # can't end with a dash
+        r"\.?"  # may have a trailing dot
+    )
+
+    ascii_only_domain_re = r"(?:\.(?!-)[a-zA-Z0-9-]{1,63}(?<!-))*"
+    ascii_only_tld_re = (
+        r"\."  # dot
+        r"(?!-)"  # can't start with a dash
+        r"(?:[a-zA-Z0-9-]{2,63})"  # domain label
+        r"(?<!-)"  # can't end with a dash
+        r"\.?"  # may have a trailing dot
+    )
+
+
+    accept_idna = True
+    max_length = 255
+
+    regex = _lazy_re_compile(domain_re + tld_re, re.IGNORECASE)
+    ascii_only_regex = _lazy_re_compile(ascii_only_domain_re + ascii_only_tld_re, re.IGNORECASE)
+
+    if not accept_idna:
```

Comment:
Having this here seems like it may be causing problems? Any logic here at the root class will not account for parameters passed at init, maybe this is blocking the accept_idna from working correctly?